src/effects/Echo.h

```diff
@@ -30,12 +30,9 @@ struct EffectEchoSettings
    double decay{ decayDefault };
 };
 
-class EffectEcho final : public EffectWithSettings<EffectEchoSettings, StatefulPerTrackEffect>
+class EffectEcho final : public EffectWithSettings<EffectEchoSettings, PerTrackEffect>
```

Comment:
The base class should change only at this step.

EffectWithSettingsBase won’t be used, so you can interactively rebase, removing the commit from me.

This step causes a MakeSettings member to be defined and called.  You must understand that it is not correct to use GetSettings until this point, because the generated MakeSettings should not be used either before this.  Only when the mSettings member of the effect is removed, must these other changes be made too, to keep all steps of the transformation correct.

Reply:
Alright, commit with EffectWithSettingsBase removed.
The other point is understood now.